_posts/2024-01-18-search-quarkus-io.adoc

```diff
@@ -0,0 +1,41 @@
+---
+layout: post
+title: 'Quarkus documentation, now with full-text search!'
+date: 2024-01-18
+tags: announcement website
+synopsis: 'We added full-text search to https://quarkus.io/guides/'
+author: yrodiere
+---
+
+:imagesdir: /assets/images/posts/l10n-of-quarkusio
+
+The Quarkus documentation has had a search feature for a long time,
+but until now it was a simple substring search on the title and summary of each guide.
+It was better than nothing, but admittedly quite limited.
+
+We've recently improved on that.
+The https://quarkus.io/guides/[guides page] now provides actual full-text search:
+
+image::../assets/images/posts/search-quarkus-io/search-vertx.png[]
+
+NOTE: The feature is not yet available on https://quarkus.io/blog/l10n-of-quarkusio/[localized sites],
+but will be in a few days, after their next sync.
+
+Among the (many!) improvements:
+
+* This is no longer a simple substring search, but state-of-the-art full-text search,
+  with tokenization and text analysis (see how `vertx` matched `Vert.x` above?).
+* Results are sorted by relevance,
+  thanks to the ability of full-text to assign a score to each hit.
+* Search takes into account the title and summary of each guide,
+  but also the _full content of the guide's HTML page_.
+* Matching terms get highlighted in the list of results,
+  giving a glimpse of relevant content from each guide.
+
+And the best part? This new search is backed by a https://github.com/quarkusio/search.quarkus.io[Quarkus app]!
+It uses in particular the https://quarkus.io/guides/hibernate-search-orm-elasticsearch[Hibernate Search extension],
+which provides https://docs.jboss.org/hibernate/stable/search/reference/en-US/html_single/#elasticsearch-integration[integration with OpenSearch/Elasticsearch].
+
+Stay tuned for more details in the coming weeks as we publish more blog posts to dive into the implementation details.
+In the meantime, happy searching!
+And feel free to drop by in the comments below if you notice weird behavior that you'd like to see fixed.
```

Comment:
Should it be either `a behavior` or `behaviors`?
```suggestion
And feel free to drop by in the comments below if you notice weird behavior that you'd like to see fixed.
```

Reply:
"Behavior" is uncountable: https://www.oxfordlearnersdictionaries.com/definition/american_english/behavior

That being said, if you think it sounds better... :shrug:

Reply:
That's not exactly true since you are in the second case where it can be countable. I think it would be better but won't fight for it!

Let's merge it :).